MetaMask/metamask-extension
# Consume Decrypt Message Manager from @metamask/message-manager

## Explanation
This PR aims to move core `DecryptMessageManager` logic into core, consume `DecryptMessageManager` and create it's own `DecryptMessageController` in extension.

For more info visit: 
https://github.com/MetaMask/MetaMask-planning/issues/415
https://github.com/MetaMask/MetaMask-planning/issues/393

`@metamask/message-manager` PR:
https://github.com/MetaMask/core/pull/1149

## Manual Testing Steps
1. Go test-dapp
2. Scroll down to "Encrypt / Decrypt"
3. Get Encrytiontion Key - Accept Confirmation 
4. Put message to encrypt - Click Encrypt
5. Click Decrypt (This is the step refactored)
6.a. If you cancel it should close Confirmation
6.b. If you accept it should show the text you originally encrypt in the box below 


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
